bin/_test-helpers.sh

```diff
@@ -0,0 +1,318 @@
+#!/bin/bash
+
+# Override CI's `set -e` default, so we can catch errors manually and display
+# proper messages
+set +e
+
+##### Test setup helpers #####
+
+export test_names=(upgrade helm helm-upgrade uninstall deep external-issuer)
```

Comment:
isn't `cluster-domain` missing from this list?

Reply:
Yep this on purpose right now, but I realize it affects the output for available tests when running `bin/tests --help`.

It's missing the same way it is currently missing in `bin/test-run`; it's used only on cloud integration where the cluster hasn't been configured with a custom cluster domain.

We use it in the kind integration tests where we can specify it with `bin/tests --test cluster-domain` and make sure it runs on a cluster that has been configured properly.

I'll change it so that `cluster-domain` renders in the help output, but it shouldn't run on the cloud integration tests.

Reply:
You meant it's only used in `kind_integration.yml` right? Might also be valuable to be available in local testing as well.

Reply:
Yep only used in `kind_integration.yml`.

I'd like to avoid adding more flags to `bin/tests`, but maybe makes sense to somehow indicate to run (or not run) certain tests? `cloud_integration.yml` would then use that to indicate it should *not* run `cluster-domain` test.